pkg/apis/serving/v1beta1/configuration_lifecycle.go

```diff
@@ -16,9 +16,30 @@ limitations under the License.
 
 package v1beta1
 
-import "k8s.io/apimachinery/pkg/runtime/schema"
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+)
+
+var configCondSet = apis.NewLivingConditionSet()
 
 // GetGroupVersionKind returns the GroupVersionKind.
-func (r *Configuration) GetGroupVersionKind() schema.GroupVersionKind {
+func (*Configuration) GetGroupVersionKind() schema.GroupVersionKind {
 	return SchemeGroupVersion.WithKind("Configuration")
 }
+
+// IsReady returns if the configuration is ready to serve the requested
+// configuration and the configuration resource has been observed.
+func (c *Configuration) IsReady() bool {
+	cs := c.Status
+	return cs.ObservedGeneration == c.Generation &&
+		cs.GetCondition(ConfigurationConditionReady).IsTrue()
```

Comment:
could use the condition set for the happy case?

ie.
https://github.com/knative/serving/blob/d21a63d235473e1a2f04f19dda35ce32bce9fc16/pkg/apis/serving/v1/configuration_lifecycle.go#L39

Reply:
I updated it from that because we don't really need to involve the whole condition manager for just reading the top-level condition (although it is useful when writing). This alternative is more directly readable ("Happiness" of conditions is all confusing to outsiders)